python/tank/bootstrap/resolver.py

```diff
@@ -381,6 +338,81 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
                 else:
                     log.debug("Pipeline configuration's 'path' fields are not set: %s" % pc)
 
+    def _create_config_descriptor(self, sg_connection, pc):
```

Comment:
see if you can encapsulate the shotgun find() call inside this method and instead pass a pipeline config id. This will provide increased robustness.